### PR TITLE
RA-1100 Fixed FindPatientByNameTest to conform with RA-980 guidelines

### PR DESCRIPTION
**Pull request by yancmo**
Ticket: https://issues.openmrs.org/browse/RA-1100

- Merged FindPatientByFamilyNameTest and FindPatientByIdAndNameTest to FindPatientByName (ID is already handled in FindPatientRecordTest)
- Added method to return FindPatientPage from HomePage

### CODE REVIEW

File: ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+package org.openmrs.reference;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.page.*;
+import org.openmrs.uitestframework.test.TestBase;
+import org.openmrs.uitestframework.test.TestData;
+import org.junit.*;
```
Comment (dkayiwa) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java]: We do not encourage using wild card imports as per: https://wiki.openmrs.org/display/docs/Coding+Conventions

**Comment by yancmo** [comment]
Updated the PR

File: ui-tests/src/test/java/org/openmrs/reference/FindPatientByIdAndNameTest.java [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-
-public class FindPatientByIdAndNameTest extends TestBase {
-    private HomePage homePage;
-    private HeaderPage headerPage;
-    private FindPatientPage findPatientPage;
-    private TestPatient patient;
-    private RegistrationPage registrationPage;
-    private ClinicianFacingPatientDashboardPage patientDashboardPage;
-    private String id;
-
```
Comment (dkayiwa) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByIdAndNameTest.java]: What is the reason behind removing this class?

Comment (yancmo) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByIdAndNameTest.java]: There are currently 2 ways to find patients: ID and name, which includes given, middle and family. Currently, there's a test which searches patients using ID, FindPatientRecordTest. The file I've added, FindPatientByNameTest searches patients by its given, middle and family name, which covers all the use cases for finding patients.
The test I'm removing searches for patients using its id, which is already covered in FindPatientRecordTest, and is one of the running tests in CI.

**Comment by yancmo** [comment]
By the way, I've made a minor change. Extended ReferenceApplicationTestBase instead of the old TestBase.

**Comment by dkayiwa** [comment]
@yancmo can you squash these commits into one as advised at?https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by yancmo** [comment]
Squashed the commits!

File: ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Test
+    @Category(org.openmrs.reference.groups.BuildTests.class)
+    public void findPatientByIdAndNameTest() throws Exception {
+        findPatientPage = homePage.clickOnFindPatientRecord();
+        // Search by name
+        findPatientPage.enterPatient(patient.givenName);
+        findPatientPage.waitForResultTable();
+        assertThat(findPatientPage.findFirstPatientName(), containsString(patient.givenName));
+
```
Comment (dkayiwa) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java]: What are the replacements for clickOnGenderLink, clickOnBirthDateLink, clickOnContactInfo, clickOnConfirmSection, etc?

Comment (yancmo) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java]: Not really sure what you mean by that, this test does not click on the patients. It just goes to the findpatient page from homepage, and enter the names into the textbox.

Comment (dkayiwa) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java]: @yancmo the test you removed was doing what i mentioned. Did you notice that?

Comment (yancmo) [inline ui-tests/src/test/java/org/openmrs/reference/FindPatientByNameTest.java]: Oh, right. For that test, the user registers a hard coded patient through the registration page (thus you'll see the methods you mentioned), before finding him in the records page. This is handled by createTestPatient in my test.

**Comment by dkayiwa** [comment]
@yancmo can you edit the ticket description by stating exactly what needs to be fixed?

**Pull request merged**